File: app/src/main/java/ua/edu/deanoffice/mobile/studentchdtu/service/client/interfaces/PostRequest.java
package ua.edu.deanoffice.mobile.studentchdtu.service.client.interfaces;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.Query;
import ua.edu.deanoffice.mobile.studentchdtu.mobile.UserData.Credentials;

public interface PostRequest {
    @Headers({"Content-Type: application/json", "Accept: */*",
    "Accept-Encoding: gzip, deflate, br"})
    @POST("api/authenticate")
    Call<ResponseBody> request(@Body Credentials cred);
}

File: app/src/main/java/ua/edu/chdtu/deanoffice/mobile/service/client/requests/interfaces/PostRequest.kt
package ua.edu.chdtu.deanoffice.mobile.service.client.requests.interfaces

import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST
import ua.edu.chdtu.deanoffice.mobile.service.POJO.ApplicationTypeIdPOJO

interface PostRequest {
    @POST("make-application")
    fun postRequest(@Body ApplicationTypeId: ApplicationTypeIdPOJO) : Call<ResponseBody>
}